compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Utils/AMDAIETransactionBuilder.cpp

```diff
@@ -97,90 +97,113 @@ LogicalResult TransactionBuilder::appendDmaStartOp(
+        enableNextBd ? std::optional<uint8_t>{static_cast<uint8_t>(
+                           *dmaBdOp.getNextBdId())}
+                     : std::nullopt;
+    if (failed(configureDMABD(
+            deviceModel, dmaDesc.value(), tileLoc, validBd,
+            static_cast<uint8_t>(*maybeBdId), enableNextBd, nextBdId,
+            enablePacket, maybePacketType, maybePacketID, maybeOutOfOrderBdId,
+            *baseAddr, dmaBdOp.getLenInBytes(), dmaBdOp.getOffsetInBytes(),
+            dmaBdOp.getBufferElementTypeWidthInBytes(), maybeDims, maybePadDims,
+            maybeIter))) {
+      return failure();
     }
-  }
-  // Disable acquire and release locks if not set.
-  if (!acqLockId) {
-    acqLockId = 0;
-    acqValue = 0;
-  }
-  if (!relLockId) {
-    relLockId = 0;
-    relValue = 0;
-  }
-  assert(acqValue && relValue && acqLockId && relLockId &&
-         "expected both use_lock(acquire) and use_lock(release) with bd");
-  if (failed(configureDMALocks(deviceModel, dmaDesc.value(), tileLoc, *acqValue,
-                               *relValue, *acqLockId, *relLockId,
-                               /*acqEn=*/true))) {
-    return failure();
-  }
-  // Pull metadata related to packet routing, bdId, buffer length, size,
-  // stride to pass to aie-rt.
-  std::optional<uint32_t> bdId = dmaBdOp.getBdId();
-  if (!bdId) return failure();
-  bool validBd = true;
-  std::optional<uint8_t> packetType;
-  std::optional<uint8_t> packetID;
-  bool enablePacket = false;
-
-  auto bufferOp = dmaBdOp.getBuffer().getDefiningOp<AMDAIE::BufferOp>();
-  if (!bufferOp) return failure();
-  std::optional<uint32_t> baseAddr = bufferOp.getAddress();
-  if (!baseAddr) return failure();
-
-  std::optional<llvm::ArrayRef<BDDimLayoutAttr>> dims = dmaBdOp.getDimensions();
-  if (!dims) return failure();
-  std::optional<SmallVector<BDDimLayout>> maybeDims;
-  maybeDims = llvm::map_to_vector(*dims, [](BDDimLayoutAttr attr) {
-    return BDDimLayout{attr.getSize(), attr.getStride()};
-  });
-  std::optional<SmallVector<BDPadLayout>> maybePadDims;
-
-  bool enableNextBd = dmaBdOp.getNextBdId().has_value();
-  std::optional<uint8_t> nextBdId =
-      enableNextBd
-          ? std::optional<uint8_t>{static_cast<uint8_t>(*dmaBdOp.getNextBdId())}
-          : std::nullopt;
-  std::optional<BDIterLayout> maybeIter = std::nullopt;
-  if (failed(configureDMABD(deviceModel, dmaDesc.value(), tileLoc, validBd,
-                            static_cast<uint8_t>(*bdId), enableNextBd, nextBdId,
-                            enablePacket, packetType, packetID, *baseAddr,
-                            dmaBdOp.getLenInBytes(), dmaBdOp.getOffsetInBytes(),
-                            dmaBdOp.getBufferElementTypeWidthInBytes(),
-                            maybeDims, maybePadDims, maybeIter))) {
-    return failure();
   }
 
   // Configure push to BD queue.
   // TODO: Generalize it as this is currently hardcoded to only shim side for
   // now.
+  AMDAIE::DMABDOp dmaBdOp =
```

Comment:
It would be good to check that `dmaStartOp.getBody().getOps<AMDAIE::DMABDOp>()` is not empty.